PayloadPkg/FirmwareUpdate/CmdFwUpdate.c

```diff
@@ -0,0 +1,161 @@
+/** @file
+  This file contains implementation of Firmware Command update library.
+
+  Copyright (c) 2020, Intel Corporation. All rights reserved.<BR>
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <Base.h>
+#include <PiPei.h>
+#include <Uefi/UefiBaseType.h>
+#include <Library/DebugLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/LitePeCoffLib.h>
+#include <Library/FirmwareUpdateLib.h>
+#include <Library/StringSupportLib.h>
+
+
+#define FLASH_DESCRIPTOR_LOCK_STR     "FLASHDESCLOCK"
+
+typedef UINT32 CMDI_TYPE;
+#define CMDI_TYPE_SPI_DESCRIPTOR_LOCK       BIT0
+
+typedef struct {
+  // Commands requested
+  CMDI_TYPE   CmdRequested;
+  // Flash Descriptor Lock command Data
+  CHAR8        *FlashDescLockDataBuf;
+  // Flash Descriptor Lock command Data length
+  UINT32      FlashDescLockDataLen;
+} CMDI_DATA;
+
+
+/**
+  Find payload in the capsule image.
+
+  This function will parse through the capsule image to find the payload
+  matching the input guid.
+
+  This function if provided with an empty guid will return the first payload
+  found
+
+  @param[in] Signature      Signature of component to update.
+  @param[in] CapImage       Pointer to the capsule Image
+  @param[in] CapImageSize   Size of the capsule image in bytes
+  @param[in] ImageHdr       Pointer to the capsule Image header
+
+  @retval  EFI_SUCCESS      Found matching payload in the capsule.
+  @retval  EFI_NOT_FOUND    No matching payload found in the capsule.
+**/
+EFI_STATUS
+FindImage (
+  IN  UINT64                        Signature,
+  IN  UINT8                         *CapImage,
+  IN  UINT32                        CapImageSize,
+  OUT EFI_FW_MGMT_CAP_IMAGE_HEADER  **ImageHdr
+  );
+
+/*
+ Parse firmware update command data buffer
+
+ Command format:
+ {FLASHDESCLOCK}
+
+  @param[in]  Buffer      Command buffer.
+  @param[in]  BufLength   Command buffer length
+  @param[out] CmdiData    Command data parsed
+*/
+EFI_STATUS
+ParseCommandStringBuf(
+  IN  UINT8       *Buffer,
+  IN  UINTN       BufLength,
+  OUT CMDI_DATA   *CmdiData
+  )
+{
+  CHAR8      *CurrLine;
+  CHAR8      *NextLine;
+  CHAR8      *EndLine;
+  CHAR8      *EndString;
+  UINT32      LineLen;
+  EFI_STATUS  Status;
+
+  Status = EFI_SUCCESS;
+  CurrLine = (CHAR8 *) Buffer;
+
+  while ((CurrLine != NULL) && (CurrLine < (Buffer + BufLength))) {
+    NextLine = GetNextLine (CurrLine, &LineLen);
+    EndLine  = CurrLine + LineLen;

```

Comment:
Are we using EndLine variable?

Reply:
EndLine is not utilized. I would remove it.